golos.publication/golos.publication.cpp

```diff
@@ -477,12 +449,13 @@ void publication::close_message(structures::mssgid message_id) {
     payto(message_id.author, eosio::asset(token_payout, state.funds.symbol), static_cast<enum_t>(payment_t::TOKEN));
     payto(message_id.author, eosio::asset(payout - token_payout, state.funds.symbol), static_cast<enum_t>(payment_t::VESTING));
 
-    //message_table.erase(mssg_itr);
-    message_index.modify(mssg_itr, _self, [&]( auto &item) {
-        item.rewardweight = static_cast<base_t>(reward_weight.data()); //not used to calculate rewards, stored for stats only
-        item.closed = true;
-        send_postclose_event(message_id.author, item);
-    });
+    tables::vote_table vote_table(_self, message_id.author.value);
+    auto votetable_index = vote_table.get_index<"messageid"_n>();
+    auto vote_itr = votetable_index.lower_bound(mssg_itr->id);
+    while ((vote_itr != votetable_index.end()) && (vote_itr->message_id == mssg_itr->id))
+        vote_itr = votetable_index.erase(vote_itr);
```

Comment:
This code is very slow, because it has copying of iterators, each copy will create new cursor to DB.
Better implementation is:

```
for (auto vote_etr = votetable_index.end(); vote_itr != vote_etr;) {
   auto& vote = *vote_itr;
   ++vote_itr;
   if (vote.message_id != mssg_itr->id) break;
   votetable_index.erase(vote);
}

```